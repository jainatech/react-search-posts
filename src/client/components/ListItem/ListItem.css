.card-text { 
    overflow: hidden;
    cursor: pointer;
    max-height: 75px;
    min-height: 75px;

    display: -webkit-box;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
    text-overflow: ellipsis;
}
.card-title{
    overflow: hidden;
    cursor: pointer;
    max-height: 25px;
    display: -webkit-box;
    -webkit-line-clamp: 1;
    -webkit-box-orient: vertical;
    text-overflow: ellipsis;
}