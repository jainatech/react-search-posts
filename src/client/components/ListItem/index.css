.post-item {
    text-align: left;
    border: solid 1px black;
    width: 100%;
  }