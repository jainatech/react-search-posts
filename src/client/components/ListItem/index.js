import React from 'react'
import PropTypes from 'prop-types'
import ENUMS from "../../utils/ENUMS";
import { Link } from 'react-router-dom';
import { Button } from 'react-bootstrap';

function ListItem({ item, index, type }) {
    switch (type) {
        case "home_posts_list":
            return (
                <tr>
                    <td>
                        <span>{item.id}.</span> <span></span>{item.title}
                    </td>
                    <td>
                        <Link to={`${ENUMS.ROUTES.EDIT_POSTS}/${item.id}`}>
                            <Button className="ml-2">Edit</Button>
                        </Link> 
                    </td>
                </tr>
            )
        default:
            return <p>List item</p>
    }


}

ListItem.propTypes = {
    item: PropTypes.object,
    index: PropTypes.number,
    type: PropTypes.string
}

export default ListItem

