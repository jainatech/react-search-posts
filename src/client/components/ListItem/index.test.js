import React from 'react';
import { mount } from 'enzyme';
import ListItem from './index';
import { BrowserRouter as Router } from 'react-router-dom';



const createTestProps = (props) => ({
  type : 'home_posts_list',
  ...props
})

describe('ListItem module case--->', () => {
  it('ListItem match snapshot', () => {
    let props = createTestProps({
        item : { "userId": 1, "id": 1, "title": "title1", "body": "body1" },
        index : 1
    })
    let wrapper = mount(<Router><ListItem {...props}></ListItem></Router>);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});