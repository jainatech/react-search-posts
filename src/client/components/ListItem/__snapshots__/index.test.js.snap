// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem module case---> ListItem match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ListItem
      index={1}
      item={
        Object {
          "body": "body1",
          "id": 1,
          "title": "title1",
          "userId": 1,
        }
      }
      type="home_posts_list"
    >
      <tr>
        <td>
          <span>
            1
            .
          </span>
           
          <span />
          title1
        </td>
        <td>
          <Link
            to="/edit/1"
          >
            <LinkAnchor
              href="/edit/1"
              navigate={[Function]}
            >
              <a
                href="/edit/1"
                onClick={[Function]}
              >
                <Button
                  active={false}
                  className="ml-2"
                  disabled={false}
                  variant="primary"
                >
                  <button
                    className="ml-2 btn btn-primary"
                    disabled={false}
                    type="button"
                  >
                    Edit
                  </button>
                </Button>
              </a>
            </LinkAnchor>
          </Link>
        </td>
      </tr>
    </ListItem>
  </Router>
</BrowserRouter>
`;
